Python packages: major updates

FRidh opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
python.pkgs.sip: 4.19.6 -> 4.19.8

Mic92 committed:
python.pkgs.pystemmer: fix compilation with python3.7

domenkozar committed:
pipenv: 2018.5.18 -> 2018.7.1

Signed-off-by: Domen Kožar <domen@dev.si>

FRidh committed:
python.pkgs.appnope: move expression

FRidh committed:
python.pkgs.audiotools: move expression

FRidh committed:
python.pkgs.av: move expression

FRidh committed:
python.pkgs.suseapi: move expression

FRidh committed:
python.pkgs.moreItertools: remove alias, use more-itertools

FRidh committed:
python.pkgs.ghp-import: move to all-packages.nix

because it's an application

Commit:
python.pkgs.pyyaml: 3.12 -> 3.13

FRidh committed:
python: adal: 1.0.1 -> 1.0.2

FRidh committed:
python: aioconsole: 0.1.8 -> 0.1.9

FRidh committed:
python: alembic: 0.9.10 -> 1.0.0

FRidh committed:
python: apipkg: 1.4 -> 1.5

FRidh committed:
python: Arpeggio: 1.8.0 -> 1.9.0

FRidh committed:
python: astor: 0.6.2 -> 0.7.1

FRidh committed:
python: astroid: 1.6.5 -> 2.0.1

FRidh committed:
python: av: 0.2.4 -> 0.4.1

FRidh committed:
python: aws-sam-translator: 1.6.0 -> 1.6.1

FRidh committed:
python: aws-xray-sdk: 1.1.1 -> 1.1.2

FRidh committed:
python: backports.lzma: 0.0.9 -> 0.0.13

FRidh committed:
python: backports.unittest_mock: 1.3 -> 1.4

FRidh committed:
python: billiard: 3.5.0.3 -> 3.5.0.4

FRidh committed:
python: biopython: 1.71 -> 1.72

FRidh committed:
python: botocore: 1.10.57 -> 1.10.62

FRidh committed:
python: zc.buildout: 2.11.5 -> 2.12.1

FRidh committed:
python: bumps: 0.7.8 -> 0.7.10

FRidh committed:
python: buttersink: 0.6.8 -> 0.6.9

FRidh committed:
python: python-can: 2.1.0 -> 2.2.1

FRidh committed:
python: cement: 2.10.2 -> 2.10.12

FRidh committed:
python: chainer: 4.1.0 -> 4.3.0

FRidh committed:
python: cheroot: 6.3.2 -> 6.3.3

FRidh committed:
python: CherryPy: 16.0.2 -> 17.0.0

FRidh committed:
python: click-log: 0.2.1 -> 0.3.2

FRidh committed:
python: cmd2: 0.9.1 -> 0.9.3

FRidh committed:
python: confluent-kafka: 0.11.4 -> 0.11.5

FRidh committed:
python: python-consul: 1.0.1 -> 1.1.0

FRidh committed:
python: cryptography: 2.2.2 -> 2.3

FRidh committed:
python: cryptography_vectors: 2.2.2 -> 2.3

FRidh committed:
python: cupy: 4.1.0 -> 4.3.0

FRidh committed:
python: cx_Oracle: 6.3.1 -> 6.4.1

FRidh committed:
python: dask: 0.18.0 -> 0.18.1

FRidh committed:
python: dbf: 0.97.7 -> 0.97.11

FRidh committed:
python: deprecation: 2.0.4 -> 2.0.5

FRidh committed:
python: dill: 0.2.8.1 -> 0.2.8.2

FRidh committed:
python: Django: 2.0.6 -> 2.0.7

FRidh committed:
python: django-sites: 0.9 -> 0.10

FRidh committed:
python: docker-compose: 1.21.2 -> 1.22.0

FRidh committed:
python: docker: 3.4.0 -> 3.4.1

FRidh committed:
python: dogpile.cache: 0.6.5 -> 0.6.6

FRidh committed:
python: dropbox: 8.9.0 -> 9.0.0

FRidh committed:
python: dulwich: 0.19.3 -> 0.19.5

FRidh committed:
python: Faker: 0.8.16 -> 0.8.17

FRidh committed:
python: Fiona: 1.7.12 -> 1.7.13

FRidh committed:
python: flake8-import-order: 0.17.1 -> 0.18

FRidh committed:
python: Flask-PyMongo: 0.5.2 -> 2.0.1

FRidh committed:
python: gensim: 3.4.0 -> 3.5.0

FRidh committed:
python: gevent: 1.3.4 -> 1.3.5

FRidh committed:
python: gitdb2: 2.0.3 -> 2.0.4

FRidh committed:
python: GitPython: 2.1.9 -> 2.1.11

FRidh committed:
python: GooCalendar: 0.3 -> 0.4

FRidh committed:
python: google-api-python-client: 1.7.3 -> 1.7.4

FRidh committed:
python: google-cloud-speech: 0.34.0 -> 0.35.0

FRidh committed:
python: greenlet: 0.4.13 -> 0.4.14

FRidh committed:
python: gssapi: 1.5.0 -> 1.5.1

FRidh committed:
python: gunicorn: 19.8.1 -> 19.9.0

FRidh committed:
python: hdbscan: 0.8.13 -> 0.8.15

FRidh committed:
python: howdoi: 1.1.7 -> 1.1.13

FRidh committed:
python: hvac: 0.6.0 -> 0.6.2

FRidh committed:
python: idna-ssl: 1.0.1 -> 1.1.0

FRidh committed:
python: ipyparallel: 6.2.1 -> 6.2.2

FRidh committed:
python: ipywidgets: 7.2.1 -> 7.3.0

FRidh committed:
python: jdatetime: 2.1.0 -> 2.2.0

FRidh committed:
python: jedi: 0.12.0 -> 0.12.1

FRidh committed:
python: joblib: 0.11 -> 0.12.1

FRidh committed:
python: jsonrpc-async: 0.6 -> 1.0.0

FRidh committed:
python: jsonrpc-base: 1.0 -> 1.0.1

FRidh committed:
python: jsonrpc-websocket: 0.6 -> 1.0.0

FRidh committed:
python: jupyterlab_launcher: 0.10.5 -> 0.11.2

FRidh committed:
python: keyring: 12.2.1 -> 13.2.1

FRidh committed:
python: ldappool: 2.2.0 -> 2.3.0

FRidh committed:
python: ledgerblue: 0.1.17 -> 0.1.19

FRidh committed:
python: libagent: 0.9.8 -> 0.11.3

FRidh committed:
python: llvmlite: 0.23.2 -> 0.24.0

FRidh committed:
python: logilab-common: 1.4.1 -> 1.4.2

FRidh committed:
python: lxml: 4.2.1 -> 4.2.3

FRidh committed:
python: m2r: 0.1.14 -> 0.1.15

FRidh committed:
python: mozinfo: 0.9 -> 0.10

FRidh committed:
python: mozlog: 3.7 -> 3.8

FRidh committed:
python: mozterm: 0.1.0 -> 1.0.0

FRidh committed:
python: msgpack-numpy: 0.4.3 -> 0.4.3.1

FRidh committed:
python: mutagen: 1.40.0 -> 1.41.0

FRidh committed:
python: mypy: 0.610 -> 0.620

FRidh committed:
python: mysqlclient: 1.3.12 -> 1.3.13

FRidh committed:
python: natsort: 5.3.2 -> 5.3.3

FRidh committed:
python: ncclient: 0.5.3 -> 0.6.0

FRidh committed:
python: nipype: 1.0.4 -> 1.1.0

FRidh committed:
python: notebook: 5.5.0 -> 5.6.0

FRidh committed:
python: numba: 0.38.1 -> 0.39.0

FRidh committed:
python: numexpr: 2.6.5 -> 2.6.6

FRidh committed:
python: numpy-stl: 2.4.1 -> 2.7.0

FRidh committed:
python: outcome: 0.1.0a0 -> 0.1.0

FRidh committed:
python: pandas: 0.23.1 -> 0.23.3

FRidh committed:
python: paramz: 0.9.1 -> 0.9.2

FRidh committed:
python: parso: 0.2.1 -> 0.3.1

FRidh committed:
python: pbr: 4.0.4 -> 4.1.1

FRidh committed:
python: phonenumbers: 8.9.8 -> 8.9.10

FRidh committed:
python: Pillow: 5.1.0 -> 5.2.0

FRidh committed:
python: pip: 10.0.1 -> 18.0

FRidh committed:
python: plaster_pastedeploy: 0.5 -> 0.6

FRidh committed:
python: plotly: 2.7.0 -> 3.1.0

FRidh committed:
python: preshed: 1.0.0 -> 1.0.1

FRidh committed:
python: progressbar: 2.3 -> 2.5

FRidh committed:
python: prometheus_client: 0.2.0 -> 0.3.0

FRidh committed:
python: ptyprocess: 0.5.2 -> 0.6.0

FRidh committed:
python: py3exiv2: 0.3.0 -> 0.4.0

FRidh committed:
python: py3status: 3.8 -> 3.11

FRidh committed:
python: pyasn1-modules: 0.2.1 -> 0.2.2

FRidh committed:
python: pycodestyle: 2.3.1 -> 2.4.0

FRidh committed:
python: pycryptodome: 3.6.2 -> 3.6.4

FRidh committed:
python: pycryptodomex: 3.6.2 -> 3.6.4

FRidh committed:
python: py: 1.5.3 -> 1.5.4

FRidh committed:
python: pyflakes: 1.6.0 -> 2.0.0

FRidh committed:
python: pygame: 1.9.3 -> 1.9.4

FRidh committed:
python: pygit2: 0.26.4 -> 0.27.1

FRidh committed:
python: pyhomematic: 0.1.44 -> 0.1.46

FRidh committed:
python: pykdtree: 1.3.0 -> 1.3.1

FRidh committed:
python: pylint: 1.9.2 -> 2.0.0

FRidh committed:
python: pylru: 1.0.9 -> 1.1.0

FRidh committed:
python: pymc3: 3.4.1 -> 3.5

FRidh committed:
python: pymongo: 3.6.1 -> 3.7.1

FRidh committed:
python: pyperclip: 1.6.2 -> 1.6.4

FRidh committed:
python: pytest: 3.6.2 -> 3.6.3

FRidh committed:
python: pytest-server-fixtures: 1.3.0 -> 1.3.1

FRidh committed:
python: python-axolotl-curve25519: 0.1 -> 0.4.1.post2

FRidh committed:
python: python-axolotl: 0.1.39 -> 0.1.42

FRidh committed:
python: python-fedora: 0.9.0 -> 0.10.0

FRidh committed:
python: python_openzwave: 0.4.4 -> 0.4.5

FRidh committed:
python: python-telegram-bot: 9.0.0 -> 10.1.0

FRidh committed:
python: pytools: 2018.4 -> 2018.5.2

FRidh committed:
python: pytz: 2018.4 -> 2018.5

FRidh committed:
python: pyzmq: 17.0.0 -> 17.1.0

FRidh committed:
python: quantities: 0.12.1 -> 0.12.2

FRidh committed:
python: regex: 2018.06.21 -> 2018.07.11

FRidh committed:
python: reikna: 0.6.8 -> 0.7.0

FRidh committed:
python: reportlab: 3.4.0 -> 3.5.1

FRidh committed:
python: requests-mock: 1.5.0 -> 1.5.2

FRidh committed:
python: rlp: 0.6.0 -> 1.0.1

FRidh committed:
python: salmon-mail: 3.0.1 -> 3.0.2

FRidh committed:
python: sarge: 0.1.4 -> 0.1.5.post0

FRidh committed:
python: scikit-learn: 0.19.1 -> 0.19.2

FRidh committed:
python: Scrapy: 1.5.0 -> 1.5.1

FRidh committed:
python: seaborn: 0.8.1 -> 0.9.0

FRidh committed:
python: seekpath: 1.8.1 -> 1.8.2

FRidh committed:
python: setuptools: 39.2.0 -> 40.0.0

FRidh committed:
python: Shapely: 1.6.4.post1 -> 1.6.4.post2

FRidh committed:
python: simplejson: 3.15.0 -> 3.16.0

FRidh committed:
python: sleekxmpp: 1.3.1 -> 1.3.3

FRidh committed:
python: smart_open: 1.5.7 -> 1.6.0

FRidh committed:
python: smmap2: 2.0.3 -> 2.0.4

FRidh committed:
python: spacy: 2.0.9 -> 2.0.12

FRidh committed:
python: spglib: 1.10.3.65 -> 1.10.3.75

FRidh committed:
python: Sphinx: 1.7.5 -> 1.7.6

FRidh committed:
python: SQLAlchemy: 1.2.8 -> 1.2.10

FRidh committed:
python: sqlmap: 1.2.6 -> 1.2.7

FRidh committed:
python: stevedore: 1.28.0 -> 1.29.0

FRidh committed:
python: stripe: 1.82.2 -> 2.0.3

FRidh committed:
python: tempora: 1.9 -> 1.13

FRidh committed:
python: textacy: 0.6.1 -> 0.6.2

FRidh committed:
python: texttable: 1.3.1 -> 1.4.0

FRidh committed:
python: tifffile: 0.14.0 -> 0.15.0

FRidh committed:
python: tornado: 5.0.2 -> 5.1

FRidh committed:
python: tox: 3.0.0 -> 3.1.2

FRidh committed:
python: treq: 17.8.0 -> 18.6.0

FRidh committed:
python: trezor: 0.9.1 -> 0.10.2

FRidh committed:
python: trio: 0.4.0 -> 0.5.0

FRidh committed:
python: Twisted: 18.4.0 -> 18.7.0

FRidh committed:
python: txtorcon: 18.0.0 -> 18.0.2

FRidh committed:
python: typeguard: 2.1.4 -> 2.2.0

FRidh committed:
python: uproot: 2.8.33 -> 2.9.6

FRidh committed:
python: vcrpy: 1.12.0 -> 1.13.0

FRidh committed:
python: vobject: 0.9.6 -> 0.9.6.1

FRidh committed:
python: websockets: 5.0.1 -> 6.0

FRidh committed:
python: widgetsnbextension: 3.2.1 -> 3.3.0

FRidh committed:
python: WSME: 0.9.2 -> 0.9.3

FRidh committed:
python: xarray: 0.10.7 -> 0.10.8

FRidh committed:
python: xml2rfc: 2.9.8 -> 2.10.0

FRidh committed:
python: yamllint: 1.9.0 -> 1.11.1

FRidh committed:
python.pkgs.bootstrapped-pip: update pip and setuptools

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=edf4f671-43a2-4c45-84ef-499d21210a4c)

Attempted: pipenv, python

<details><summary>Partial log (click to expand)</summary><p>

```
    % (self.verifier.get_module_name()))

-- Docs: http://doc.pytest.org/en/latest/warnings.html
===== 1879 passed, 90 skipped, 4 xfailed, 202 warnings in 1031.82 seconds ======
cannot build derivation '/nix/store/w3iyim0md2hl8qbwbwnzd9mk9654axp7-python3.6-cryptography-2.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r6fwg14gwj0arz7dci3izah1c9zm415r-python3.6-pyOpenSSL-18.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/pn7xij83g9c8mgr61ml5d3jd6q43lmx5-python3.6-urllib3-1.23.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/8v8wfpmyw75w6x4axxpvx3lchihww2cr-python3.6-requests-2.19.1.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/k0nwavbycn4426h6873fvp3aib5slyd3-pipenv-2018.7.1.drv': 3 dependencies couldn't be built
error: build of '/nix/store/k0nwavbycn4426h6873fvp3aib5slyd3-pipenv-2018.7.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=758a01ba-6204-4f23-bdcc-40afa41ea966)

Attempted: pipenv, python

<details><summary>Partial log (click to expand)</summary><p>

```
    % (self.verifier.get_module_name()))

-- Docs: http://doc.pytest.org/en/latest/warnings.html
===== 1869 passed, 100 skipped, 4 xfailed, 201 warnings in 983.47 seconds ======
cannot build derivation '/nix/store/2vcgx24mp5wndwbkr5sx7f362b80zyf1-python3.6-cryptography-2.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6pf26qf7r6nqj72zsmk40kh6sag6bb8g-python3.6-pyOpenSSL-18.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/rfcyvsx8a4gmnp0gc76jhkcxp3w0n0lx-python3.6-urllib3-1.23.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/amb8aggi3c1622rahwkvcn33lpjbmw60-python3.6-requests-2.19.1.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/0mn58l7wyv8f0a4h53h3czpjpxagy66w-pipenv-2018.7.1.drv': 3 dependencies couldn't be built
error: build of '/nix/store/0mn58l7wyv8f0a4h53h3czpjpxagy66w-pipenv-2018.7.1.drv' failed
```
</p></details>

volth commented on pkgs/development/python-modules/jsonrpc-base/default.nix:
```diff
@@ -2,11 +2,11 @@
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1dl55n54ha5kf4x6hap2p1k3s4qa4w7g791wp2656rjg2zxfgywk";
+    sha256 = "21f860c915617f6475aa1ac5a1ec11de03cce6b279741f25ad97d8a4c5b76c3c";
```
any reason for base32->base16 changes?

dotlambda replied:
The update script gets the hashes from PyPI, where they are in a base16 format, see https://pypi.org/project/jsonrpc-base/#files.

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=230fbd97-474a-4bb5-9695-69b6815da4af)

Attempted: pipenv, python

<details><summary>Partial log (click to expand)</summary><p>

```
Installing collected packages: requests
Successfully installed requests-2.19.1
/private/tmp/nix-build-python3.6-requests-2.19.1.drv-0/requests-2.19.1
post-installation fixup
strip is /nix/store/sixx1lfy97125f6hxymbwn64cb8z62p3-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/lf176ifs9qj4b5rd69l9d391l7hfxprx-python3.6-requests-2.19.1/lib
patching script interpreter paths in /nix/store/lf176ifs9qj4b5rd69l9d391l7hfxprx-python3.6-requests-2.19.1
strip is /nix/store/sixx1lfy97125f6hxymbwn64cb8z62p3-cctools-binutils-darwin/bin/strip
cannot build derivation '/nix/store/m09lhmcaqzsl57sjgl44wmbc5a9pdx5h-pipenv-2018.7.1.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/m09lhmcaqzsl57sjgl44wmbc5a9pdx5h-pipenv-2018.7.1.drv' failed
```
</p></details>

FRidh committed:
python.pkgs.apipkg: fix build

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=04741e10-e89f-47b8-979b-a47c6ad8d7d9)

Attempted: pipenv, python

<details><summary>Partial log (click to expand)</summary><p>

```
Installing collected packages: requests
Successfully installed requests-2.19.1
/private/tmp/nix-build-python3.6-requests-2.19.1.drv-0/requests-2.19.1
post-installation fixup
strip is /nix/store/sixx1lfy97125f6hxymbwn64cb8z62p3-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/1bnzrzdxggg2azawbag82zyqpb938scm-python3.6-requests-2.19.1/lib
patching script interpreter paths in /nix/store/1bnzrzdxggg2azawbag82zyqpb938scm-python3.6-requests-2.19.1
strip is /nix/store/sixx1lfy97125f6hxymbwn64cb8z62p3-cctools-binutils-darwin/bin/strip
cannot build derivation '/nix/store/8qbjw7pkj9zn6f7wh7b494gpcma25j78-pipenv-2018.7.1.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/8qbjw7pkj9zn6f7wh7b494gpcma25j78-pipenv-2018.7.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=88dc49a8-e08a-4253-a05d-2d2f5dea93fc)

Attempted: pipenv, python

<details><summary>Partial log (click to expand)</summary><p>

```
/build/requests-2.19.1
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/2jqyaamav62pp5n783pcz77iqga8a2bj-python3.6-requests-2.19.1
strip is /nix/store/a3nk8z2i7m7wa3jdckgv710n7j3yx4b5-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/2jqyaamav62pp5n783pcz77iqga8a2bj-python3.6-requests-2.19.1/lib
patching script interpreter paths in /nix/store/2jqyaamav62pp5n783pcz77iqga8a2bj-python3.6-requests-2.19.1
checking for references to /build in /nix/store/2jqyaamav62pp5n783pcz77iqga8a2bj-python3.6-requests-2.19.1...
strip is /nix/store/a3nk8z2i7m7wa3jdckgv710n7j3yx4b5-binutils-2.30/bin/strip
cannot build derivation '/nix/store/795j8h44s2pwn5zacz1ndfb2lhbq3ikj-pipenv-2018.7.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/795j8h44s2pwn5zacz1ndfb2lhbq3ikj-pipenv-2018.7.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=b8b6f89c-3df8-482a-9b27-4c9444de4600)

Attempted: pipenv, python

<details><summary>Partial log (click to expand)</summary><p>

```
/build/requests-2.19.1
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/pypz6q2bjjzvkc4v7yxhmp02zsl253k4-python3.6-requests-2.19.1
strip is /nix/store/a245zacjzf3qw0davhvlfarihcy2yyrc-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/pypz6q2bjjzvkc4v7yxhmp02zsl253k4-python3.6-requests-2.19.1/lib
patching script interpreter paths in /nix/store/pypz6q2bjjzvkc4v7yxhmp02zsl253k4-python3.6-requests-2.19.1
checking for references to /build in /nix/store/pypz6q2bjjzvkc4v7yxhmp02zsl253k4-python3.6-requests-2.19.1...
strip is /nix/store/a245zacjzf3qw0davhvlfarihcy2yyrc-binutils-2.30/bin/strip
cannot build derivation '/nix/store/r1znq8kw8r2awcjld60638xzqry3dh1v-pipenv-2018.7.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/r1znq8kw8r2awcjld60638xzqry3dh1v-pipenv-2018.7.1.drv' failed
```
</p></details>

FRidh committed:
python.pkgs.pytest-flake8: disable test, fix build

Commit:
python.pkgs.flake8: allow newer versions of pycodestyle and pyflakes

This applies appropriate upstream patches from
https://gitlab.com/pycqa/flake8/commits/master/setup.py.

Commit:
python.pkgs.twisted: update propagatedBuildInputs

Commit:
python.pkgs.python-axolotl: update propagatedBuildInputs

Commit:
python.pkgs.python-axolotl: enable tests

Commit:
Revert "python: pygit2: 0.26.4 -> 0.27.1"

This reverts commit e9bba1cb01ea7e45c1051f29f33c78b9bb502008.
python.pkgs.pygit2 has to be upgraded at the same time as libgit2.

Mic92 committed:
python.pkgs.snowballstemmer: move out of python-packages.nix

Commit:
python.pkgs.node-semver: remove tox from checkInputs

Commit:
python.pkgs.tox: update propagatedBuildInputs

Commit:
python.pkgs.face_recognition: remove tox from checkInputs

Commit:
python.pkgs.ipython: allow prompt_toolkit 2.0

This applies the appropriate upstream patch:
https://github.com/ipython/ipython/commit/8e256bd37373f98580ba1ef1d3fcfd7976802238

FRidh committed:
python.pkgs.pandas: fix build

Commit:
python.pkgs.jaraco_functools: move expression

Commit:
python.pkgs.tempora: update propagatedBuildInputs

Commit:
python.pkgs.jaraco_functools: 1.15.1 -> 1.20

Commit:
python: testresources: 0.2.7 -> 2.0.1

Commit:
python.pkgs.astroid: fix build

Commit:
python.pkgs.flake8: remove no longer used patch

Commit:
conan: 1.5.2 -> 1.6.0

dotlambda commented:
@GrahamcOfBorg build conan

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=e3515243-adc2-4268-9eb7-999e2fe71daf)

The following builds were skipped because they don't evaluate on aarch64-linux: conan

<details><summary>Partial log (click to expand)</summary><p>

```
while evaluating the attribute 'astroid.overridePythonAttrs' at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/top-level/python-packages.nix:633:3:
while evaluating 'callPackageWith' at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/lib/customisation.nix:108:35, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/top-level/python-packages.nix:633:13:
while evaluating 'makeOverridable' at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/lib/customisation.nix:67:24, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/lib/customisation.nix:112:8:
while evaluating anonymous function at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/development/python-modules/astroid/default.nix:1:1, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/lib/customisation.nix:69:12:
while evaluating 'makeOverridablePythonPackage' at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/top-level/python-packages.nix:40:37, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/development/python-modules/astroid/default.nix:6:1:
while evaluating 'makeOverridable' at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/lib/customisation.nix:67:24, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/top-level/python-packages.nix:42:12:
while evaluating anonymous function at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/development/interpreters/python/build-python-package.nix:27:1, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/lib/customisation.nix:69:12:
while evaluating anonymous function at /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/development/interpreters/python/mk-python-derivation.nix:15:1, called from /var/lib/gc-of-borg/nix-test-rs-29/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-aarch64-community-29/pkgs/development/interpreters/python/build-python-package.nix:44:4:
astroid-2.0.1 not supported for interpreter python2.7

```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43972&attempt_id=6ebcb970-76c0-4bd0-b7ac-79c76c3e4330)

The following builds were skipped because they don't evaluate on x86_64-linux: conan

<details><summary>Partial log (click to expand)</summary><p>

```
while evaluating the attribute 'astroid.overridePythonAttrs' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/top-level/python-packages.nix:633:3:
while evaluating 'callPackageWith' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/lib/customisation.nix:108:35, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/top-level/python-packages.nix:633:13:
while evaluating 'makeOverridable' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/lib/customisation.nix:67:24, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/lib/customisation.nix:112:8:
while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/development/python-modules/astroid/default.nix:1:1, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/lib/customisation.nix:69:12:
while evaluating 'makeOverridablePythonPackage' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/top-level/python-packages.nix:40:37, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/development/python-modules/astroid/default.nix:6:1:
while evaluating 'makeOverridable' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/lib/customisation.nix:67:24, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/top-level/python-packages.nix:42:12:
while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/development/interpreters/python/build-python-package.nix:27:1, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/lib/customisation.nix:69:12:
while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/development/interpreters/python/mk-python-derivation.nix:15:1, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/builder-0-gustav.ewr1.nix.ci/pkgs/development/interpreters/python/build-python-package.nix:44:4:
astroid-2.0.1 not supported for interpreter python2.7

```
</p></details>

dotlambda commented on pkgs/development/tools/build-managers/conan/default.nix:
```diff
@@ -16,16 +16,24 @@ let newPython = python.override {
+        inherit version;
+        sha256 = "fc9b582dba0366e63540982c3944a9230cbc6f303641c51483fa547dcc22393a";
+      };
+    });
```
@FRidh Do you have any idea why this doesn't work?

> astroid-2.0.1 not supported for interpreter python2.7

dotlambda replied:
Seems to work when I use `python3`. So, `packageOverrides` must be broken for `python2`.

Commit:
rtv: use python3Packages

FRidh committed:
python.pkgs.pytest: fix setupHook .pytest-cache

In 610485faa7f8c043482c4a9c030938ed315557e5 a setupHook was added to `pytest`
to prevent the creation of `.pytest-cache` folder. Some pytest plugins relied
on the cache and were thus failing.

This commit permits the cache during build time but removes it in a `postFixupHook`.

Commit:
python.pkgs.importlib: remove

importlib is included in Python 2.7+ and 3.1+

Commit:
python.pkgs.can: fix build

Commit:
python.pkgs.pylint: 2.0.0 -> 2.0.1

Commit:
python.pkgs.sqlalchemy_migrate: remove pylint from checkInputs

Seems not to be needed and does not work on Python 2.

Commit:
spyder: use python3

Dependency pylint is no longer supported on Python 2.

Commit:
beets: use python3Packages

Commit:
spyder: remove from pythonPackages

Spyder is an application only and therefore does not belong in pythonPackages.

FRidh committed:
python.pkgs.pytest-flake8: disable tests

FRidh committed:
python.pkgs.prompt_toolkit_52: remove unused

FRidh committed:
python.pkgs.JPype1: disable tests

FRidh committed:
python.pkgs.nipype: add missing dependency

FRidh committed:
python: aioconsole: 0.1.9 -> 0.1.10

FRidh committed:
python: asyncssh: 1.13.2 -> 1.13.3

FRidh committed:
python: botocore: 1.10.62 -> 1.10.64

FRidh committed:
python: dask: 0.18.1 -> 0.18.2

FRidh committed:
python: IBMQuantumExperience: 1.9.6 -> 1.9.8

FRidh committed:
python: jdatetime: 2.2.0 -> 2.2.1

FRidh committed:
python: libusb1: 1.6.4 -> 1.6.5

FRidh committed:
python: pyasn1: 0.4.3 -> 0.4.4

FRidh committed:
python: pytest-timeout: 1.3.0 -> 1.3.1

FRidh committed:
python: pytest-xdist: 1.22.2 -> 1.22.3

FRidh committed:
python: python_openzwave: 0.4.5 -> 0.4.7

FRidh committed:
python: pyxattr: 0.6.0 -> 0.6.1

FRidh committed:
python: reportlab: 3.5.1 -> 3.5.2

FRidh committed:
python: seqdiag: 0.9.5 -> 0.9.6

FRidh committed:
python: tifffile: 0.15.0 -> 0.15.1

FRidh committed:
python.pkgs.scikitlearn: disable tests; remove maintainer

dotlambda commented:
@FRidh I'm sorry, I was away for a few days and forgot to revert the spyder changes daabcc3 and 48eb8c4.

FRidh commented:
@dotlambda actually I removed them and rebased. I don't think the change would have brought us anything, really. Just work for those that had to change their expression.

andreabedini commented:
This broke `aws-sam-cli` which relies on `aws-sam-translator == 1.6.0` (which this PR updates to 1.6.1). Since I contributed the initial derivation for aws-sam-cli, I'm keen to fix the build but ... how?

dotlambda commented:
@andreabedini I think the simplest thing is to add a `postPatch` hook, where you use `substituteInPlace` to replace `==1.6.0` by `>=1.6.0`. Plus, you should make upstream aware of it.

andreabedini commented:
> @andreabedini I think the simplest thing is to add a postPatch hook, where you use substituteInPlace to replace ==1.6.0 by >=1.6.0. Plus, you should make upstream aware of it.

thanks @dotlambda, I will try that
